src/banded/bidiagonalconjugationdata.jl

```diff
@@ -0,0 +1,85 @@
+"""
+    BidiagonalConjugationData{T, MU, MC} <: LazyMatrix{T}
+
+Struct for efficiently representing the matrix product `A = inv(U)XV`,
+assuming that 
+
+- `A` is upper bidiagonal, 
+- `U` is upper Hessenberg,
+- `X` is banded,
+- `V` is banded.
+
+None of these properties are checked internally. It is the user's responsibility
+to ensure these properties hold and that the product `inv(U)XV` is indeed bidiagonal.
+
+# Fields 
+- `U`: The upper Hessenberg matrix. 
+- `C`: The matrix product `XV`.
+- `dv`: A vector giving the diagonal of `A`. 
+- `ev`: A vector giving the superdiagonal of `A`.
+
+The vectors `dv` and `ev` grow on demand from `getindex` and should not be 
+used directly. Simply treat 
+
+    A = BidiagonalConjugationData(U, X, V)
+
+as you would an upper bidiagonal matrix.
+"""
+struct BidiagonalConjugationData{T,MU,MC} <: LazyMatrix{T}
```

Comment:
The `*Data` name would be if this is just a back-storage for `BidiagonalConjugationBand` which would be the vectors plugged into a `Bidiagonal`, see https://github.com/JuliaApproximation/ClassicalOrthogonalPolynomials.jl/blob/bed20a3855d404ef8f60be671bf6489633a4e750/src/lanczos.jl#L126 for an example of this setup. Since this represents the whole matrix `BidiagonalConjugation` would make more sense.